corehq/apps/sms/views.py

```diff
@@ -751,7 +751,8 @@ def chat_contact_list(request, domain):
     total_records = len(data)
 
     if sSearch:
-        regex = re.compile('^.*%s.*$' % sSearch)
+        safe_search = re.escape(sSearch)
+        regex = re.compile('^.*%s.*$' % safe_search)
```

Comment:
So, whatever text we get in sSearch, we are putting that in the regular expression.
and then we are filtering the phone numbers we have for the domain.

The PR description says that we are now limiting to just alphanumerics which does seem reasonable except for symbols like '+' or '-' that could be a part of the phone numbers and hence sSearch which will cause a change in behaviour of this end point?

Reply:
It actually only escapes non-alphanumerics, it doesn't prohibit or strip them.  If you have a `+` in the search string, for example, that is interpreted as the regex character meaning "one or more of the previous character".  Right now, if you type `+1904` in the search box, you get [a 500 error](https://dimagi.sentry.io/share/issue/0cdde36fee5d4f15a6203d21a95f9f57/), since that produces an invalid regex.  This change means that the `+` will be treated as a normal string character.

This example might help illustrate the behavior:
```python
In [1]: re.compile('^.*%s.*$' % '123-456').match('abc123-456')
Out[1]: <re.Match object; span=(0, 10), match='abc123-456'>

In [2]: re.escape('123-456')
Out[2]: '123\\-456'

In [3]: re.compile('^.*%s.*$' % re.escape('123-456')).match('abc123-456')
Out[3]: <re.Match object; span=(0, 10), match='abc123-456'>
```

You are correct that this is a change in behavior though, in that users will no longer be able to use regexs here.  The UI and documentation for the feature don't mention them, but I'll check to make sure we're okay with that change first.  Thanks for flagging.

Reply:
Okay, is there a documentation we need to update?
So we are not supporting regex anymore?

Reply:
That's right, this is a user-facing change to no longer support regex.  I/we believe this is acceptable as the page and documentation do not mention that regex is used behind the scenes, and most of our other search boxes don't use regex.  That plus the fact that code scanning tools flag this as a potential risk, seems like the best bet is just to remove the functionality (though for what it's worth, I don't think the risk is very realistic)

Reply:
That works for me. Are you good to update the description accordingly?

Reply:
good point, updated.  Thanks for the review!